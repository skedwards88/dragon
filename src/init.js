import { locations } from "./locations.js";
import { items } from "./items.js";

function buildStartingLocations() {
  const startingItemLocations = {
    inventory: [],
    outOfPlay: [],
  };

  Object.keys(locations).forEach(
    (location) => (startingItemLocations[location] = [])
  );

  for (const [item, itemInfo] of Object.entries(items)) {
    startingItemLocations[itemInfo.spawnLocation].push(item);
  }

  return startingItemLocations;
}

export function init() {
  const startingItemLocations = buildStartingLocations();

  return {
    reputation: 10,
    gold: 0,
    timeInCave: 0,
    swordCost: 40,
    maxSwordCost: 50,
    ownSword: false,
    manorFire: true,
    naked: true,
    squirrelDead: false,
    horseDead: false,
    horseTethered: false,
    horseMounted: false,
    playerPoisoned: false,
    clothesPoopy: false,
    handkerchiefDamp: false,
    playerMasked: false,
    babyCough: false,
    playerCough: false,
    savedBaby: false,
    receivedBabyReward: false,
    playedForYouth: false,
    offeredHandkerchiefToYouth: false,
    cursed: false,
    firstCourtyardEntry: true,
    dragonPoisoned: false,
    dragonAsleep: false,
    dragonDead: false,
    treasureAmount: 300,
    treasureLevel: 0,
    singeCount: 0,
    gotScoreByCredit: false,
    paidDebt: false,
    gotScoreByTrade: false,
    maxReputation: 17,
    maxGold: 310,
    appleBitesRemaining: 5,

    playerLocation: "room",

    consequenceText: "",

    locationConsequenceText: "",

    itemLocations: startingItemLocations,

    journalPagesRemaining: 3,
    journalEntry: null,
  };
}
