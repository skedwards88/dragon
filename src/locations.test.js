import { locations } from "./locations";
import { init } from "./init";

const newGameState = init();

test("Inn description, apple not taken, not clothed", () => {
  const location = "inn";
  const gameState = {
    ...newGameState,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(`
    "You enter what appears to be the common room of an inn. A complementary apple rests on the table. 

    The inn keeper laughs, "Haven't you heard of clothes?!""
  `);
});

test("Inn description, apple not taken, clothed", () => {
  const location = "inn";
  const gameState = {
    ...newGameState,
    naked: false,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You enter what appears to be the common room of an inn. A complementary apple rests on the table. "`
  );
});

test("Inn description, apple not taken but partially eaten, clothed", () => {
  const location = "inn";
  const gameState = {
    ...newGameState,
    naked: false,
    appleBitesRemaining: 4,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You enter what appears to be the common room of an inn. "`
  );
});

test("Inn description, apple taken, not clothed", () => {
  const location = "inn";
  const gameState = {
    ...newGameState,
    itemLocations: { ...newGameState.itemLocations, inn: ["lute"] },
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(`
    "You enter what appears to be the common room of an inn. 

    The inn keeper laughs, "Haven't you heard of clothes?!""
  `);
});

test("Inn description, apple not taken, clothed", () => {
  const location = "inn";
  const gameState = {
    ...newGameState,
    naked: false,
    itemLocations: { ...newGameState.itemLocations, inn: ["lute"] },
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You enter what appears to be the common room of an inn. "`
  );
});

test("Mirror description, clothed", () => {
  const location = "mirror";
  const gameState = {
    ...newGameState,
    naked: false,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You are quite good looking, if you say so yourself. "`
  );
});

test("Mirror description, not clothed", () => {
  const location = "mirror";
  const gameState = {
    ...newGameState,
    naked: true,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(`"You're naked!"`);
});

test("Wardrobe description, clothes not taken", () => {
  const location = "wardrobe";
  const gameState = {
    ...newGameState,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"Inside the wardrobe, there is a mirror and a set of clothes. "`
  );
});

test("Wardrobe description, clothes taken", () => {
  const location = "wardrobe";
  const gameState = {
    ...newGameState,
    itemLocations: { ...newGameState.itemLocations, wardrobe: ["lute"] },
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"Inside the wardrobe, there is a mirror. "`
  );
});

test("Inn window description, manor on fire", () => {
  const location = "window";
  const gameState = {
    ...newGameState,
    manorFire: true,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"Through the window, you see flames and smoke coming from a nearby mansion. A crowd has gathered in front of the mansion. "`
  );
});

test("Inn window description, manor not on fire", () => {
  const location = "window";
  const gameState = {
    ...newGameState,
    manorFire: false,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"Through the window, you see the charred remains of a nearby mansion. "`
  );
});

test("Room description, lute not taken", () => {
  const location = "room";
  const gameState = {
    ...newGameState,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You are in a bedroom with a window, wardrobe, and door. A lute leans against the bed. You smell smoke and hear screams in the distance. "`
  );
});

test("Room description, lute taken", () => {
  const location = "room";
  const gameState = {
    ...newGameState,
    itemLocations: { ...newGameState.itemLocations, room: [] },
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You are in a bedroom with a window, wardrobe, and door. You smell smoke and hear screams in the distance. "`
  );
});

test("Room description, lute not taken, no fire", () => {
  const location = "room";
  const gameState = {
    ...newGameState,
    manorFire: false,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You are in a bedroom with a window, wardrobe, and door. A lute leans against the bed. "`
  );
});

test("Room description, lute taken", () => {
  const location = "room";
  const gameState = {
    ...newGameState,
    manorFire: false,
    itemLocations: { ...newGameState.itemLocations, room: [] },
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You are in a bedroom with a window, wardrobe, and door. "`
  );
});

test("Courtyard description, fire, first time", () => {
  const location = "courtyard";
  const gameState = {
    ...newGameState,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(`
    "You are in a small courtyard that connects to the inn. You can see a fountain and hear sounds of a blacksmith shop. Beyond the fountain, you see flames and smoke. 

    A youth runs from the fire, crying as they flee. They drop a handkerchief in their distress. "
  `);
});

test("Courtyard description, fire, not first time", () => {
  const location = "courtyard";
  const gameState = {
    ...newGameState,
    firstCourtyardEntry: false,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You are in a small courtyard that connects to the inn. You can see a fountain and hear sounds of a blacksmith shop. Beyond the fountain, you see flames and smoke. "`
  );
});

test("Courtyard description, no fire, first time", () => {
  const location = "courtyard";
  const gameState = {
    ...newGameState,
    manorFire: false,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(`
    "You are in a small courtyard that connects to the inn. You can see a fountain and hear sounds of a blacksmith shop. 

    A youth runs from the fire, crying as they flee. They drop a handkerchief in their distress. "
  `);
});

test("Courtyard description, no fire, not first time", () => {
  const location = "courtyard";
  const gameState = {
    ...newGameState,
    firstCourtyardEntry: false,
    manorFire: false,
  };
  const description = locations[location].getDescription(gameState);
  expect(description).toMatchInlineSnapshot(
    `"You are in a small courtyard that connects to the inn. You can see a fountain and hear sounds of a blacksmith shop. "`
  );
});
