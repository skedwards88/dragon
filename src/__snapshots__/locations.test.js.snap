// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fountain description: naked false, babyCough false, playerCough false 1`] = `
"You stand at the edge of a fountain. In the center is a statue of a dragon and cowering people. 

You hear a voice: "My baby! You saved my baby! Please take this gold as thanks." 

Behind you, you hear the roof collapse. Finally, the crowd is able to douse the flames. "
`;

exports[`Fountain description: naked false, babyCough false, playerCough true 1`] = `
"You stand at the edge of a fountain. In the center is a statue of a dragon and cowering people. 

You hear a voice: "My baby! You saved my baby! Please take this gold as thanks." 

As you take the gold, you let out a racking cough. It seems that you should have taken measures to avoid breathing the smoke. They pass you the gold quickly, eager to get away from your coughing. 

Behind you, you hear the roof collapse. Finally, the crowd is able to douse the flames. "
`;

exports[`Fountain description: naked false, babyCough true, playerCough false 1`] = `
"You stand at the edge of a fountain. In the center is a statue of a dragon and cowering people. 

You hear a voice: "My baby! You saved my baby!" The baby coughs from the smoke. The parent glares at you, "My baby has a terrible cough from being carried through the smoke. Regardless, take this gold as thanks." 

Behind you, you hear the roof collapse. Finally, the crowd is able to douse the flames. "
`;

exports[`Fountain description: naked false, babyCough true, playerCough true 1`] = `
"You stand at the edge of a fountain. In the center is a statue of a dragon and cowering people. 

You hear a voice: "My baby! You saved my baby!" The baby coughs from the smoke. The parent glares at you, "My baby has a terrible cough from being carried through the smoke. Regardless, take this gold as thanks." 

As you take the gold, you let out a racking cough. It seems that you should have taken measures to avoid breathing the smoke. They pass you the gold quickly, eager to get away from your coughing. 

Behind you, you hear the roof collapse. Finally, the crowd is able to douse the flames. "
`;

exports[`Fountain description: naked true, babyCough false, playerCough false 1`] = `
"You stand at the edge of a fountain. In the center is a statue of a dragon and cowering people. 

You hear a voice: "My baby! You saved my baby! Please take this gold as thanks. And perhaps buy yourself some clothes," they say, eyeing your naked body. 

Behind you, you hear the roof collapse. Finally, the crowd is able to douse the flames. "
`;

exports[`Fountain description: naked true, babyCough false, playerCough true 1`] = `
"You stand at the edge of a fountain. In the center is a statue of a dragon and cowering people. 

You hear a voice: "My baby! You saved my baby! Please take this gold as thanks. And perhaps buy yourself some clothes," they say, eyeing your naked body. 

As you take the gold, you let out a racking cough. It seems that you should have taken measures to avoid breathing the smoke. They pass you the gold quickly, eager to get away from your coughing. 

Behind you, you hear the roof collapse. Finally, the crowd is able to douse the flames. "
`;

exports[`Fountain description: naked true, babyCough true, playerCough false 1`] = `
"You stand at the edge of a fountain. In the center is a statue of a dragon and cowering people. 

You hear a voice: "My baby! You saved my baby!" The baby coughs from the smoke. The parent glares at you, "My baby has a terrible cough from being carried through the smoke. Regardless, take this gold as thanks. And perhaps buy yourself some clothes," they say, eyeing your naked body. 

Behind you, you hear the roof collapse. Finally, the crowd is able to douse the flames. "
`;

exports[`Fountain description: naked true, babyCough true, playerCough true 1`] = `
"You stand at the edge of a fountain. In the center is a statue of a dragon and cowering people. 

You hear a voice: "My baby! You saved my baby!" The baby coughs from the smoke. The parent glares at you, "My baby has a terrible cough from being carried through the smoke. Regardless, take this gold as thanks. And perhaps buy yourself some clothes," they say, eyeing your naked body. 

As you take the gold, you let out a racking cough. It seems that you should have taken measures to avoid breathing the smoke. They pass you the gold quickly, eager to get away from your coughing. 

Behind you, you hear the roof collapse. Finally, the crowd is able to douse the flames. "
`;
